src/main/kotlin/racingcar/domain/RacingGame.kt

```diff
@@ -0,0 +1,32 @@
+package racingcar.domain
+
+class RacingGame(
+    private val cars: Cars,
+    private var laps: Int,
+    private val movingStrategy: MovingStrategy = RandomStrategy()
+) {
+    private val lapReports: MutableList<List<Int>> = mutableListOf()
+
+    fun start() {
+        if (!isNotEnd()) throw IllegalStateException("이미 종료된 레이스입니다.")
+        cars.move(movingStrategy)
+        writeReport()
+        laps--
+    }
+
+    private fun writeReport() {
+        lapReports.add(cars.location().map { it.value })
+    }
+
+    fun isNotEnd(): Boolean {
+        return laps > END_GAME_LAPS
+    }
+
+    fun report(): List<List<Int>> {
+        return lapReports
```

Comment:
```suggestion
        return lapReports.toList()
```
원본을 immutable로 리턴하더라도, mutable로 형변환하면 원본을 수정할 수 있어요.
방어적 복사를 활용해보면 어떨까요?

Reply:
backing 프로퍼티를 활용하면, 함수를 별도로 정의하지 않아도 돼요.

ref. https://kotlinlang.org/docs/coding-conventions.html#names-for-backing-properties